PWGJE/Tasks/jetSpectraEseTask.cxx

```diff
@@ -72,76 +78,82 @@ struct JetSpectraEseTask {
+    registry.fill(HIST("h_collisions"), 0.5);
+    if (originalCollision.fESECOL()[0] != 1)
+      return;
+    registry.fill(HIST("h_collisions"), 1.5);
+    if (originalCollision.centFT0C() < CentRange->at(0) || originalCollision.centFT0C() > CentRange->at(1))
+      return;
+    registry.fill(HIST("h_collisions"), 2.5);
     float vPsi2 = FFitWeights::EventPlane(originalCollision, 2);
-
     auto qPerc = originalCollision.qPERCFT0C();
-    if (qPerc[0] > 0)
-      registry.fill(HIST("h_Cent_q2Percs"), originalCollision.centFT0C(), qPerc[0]);
-
-    int code = spCode(qPerc[0]);
-
-    registry.fill(HIST("h_collisions"), 0.5);
     if (!jetderiveddatautilities::selectCollision(collision, eventSelection)) {
       return;
     }
+    registry.fill(HIST("h_collisions"), 3.5);
+    double leadingTrackpT = 0.0;
+    for (auto& track : tracks) {
+      if (track.pt() > 5.0) {
+        if (track.pt() > leadingTrackpT) {
+          leadingTrackpT = track.pt();
+        }
+      }
+    }
+    if (leadingTrackpT == 0.0)
+      return;
 
-    registry.fill(HIST("h_collisions"), 1.5);
-    float plane{0};
-    std::string pl{""};
+    registry.fill(HIST("h_collisions"), 4.5);
+    registry.fill(HIST("h_rho"), collision.rho());
     for (auto const& jet : jets) {
-
-      plane = vPsi2 - jet.phi();
-      pl = jetplane(plane);
+      float jetpT_bkgsub = jet.pt() - (collision.rho() * jet.area());
       registry.fill(HIST("h_jet_pt"), jet.pt());
-
-      if (code < 0)
+      registry.fill(HIST("h_jet_pt_bkgsub"), jetpT_bkgsub);
+      registry.fill(HIST("h_jet_eta"), jet.eta());
+      registry.fill(HIST("h_jet_phi"), jet.phi());
+      registry.fill(HIST("h_Psi2"), vPsi2);
+      registry.fill(HIST("h_jet_area"), jet.area());
+
+      float dPhi = DeltaRPhi(jet.phi(), vPsi2);
+      registry.fill(HIST("h_dPhi"), dPhi);
+      if (qPerc[0] < 0)
         continue;
-
-      if (code == 0)
-        registry.fill(HIST("h_jet_pt_q2_0"), jet.pt());
-      if (code == 1)
-        registry.fill(HIST("h_jet_pt_q2_1"), jet.pt());
-      if (pl == "in")
-        registry.fill(HIST("h_jet_pt_in"), jet.pt());
-      if (pl == "out")
-        registry.fill(HIST("h_jet_pt_out"), jet.pt());
+      registry.fill(HIST("jet_pt_dPhi_q2"), jetpT_bkgsub, dPhi, qPerc[0]);
     }
+    registry.fill(HIST("h_collisions"), 5.5);
+  }
+  PROCESS_SWITCH(JetSpectraEseTask, processESEDataCharged, "process ese collisions", true);
+
+  float DeltaRPhi(Double_t mphi, Double_t vphi)
+  {
+    if (vphi < -1 * TMath::Pi())
+      vphi += (2 * TMath::Pi());
+    else if (vphi > TMath::Pi())
+      vphi -= (2 * TMath::Pi());
+
+    if (mphi < -1 * TMath::Pi())
+      mphi += (2 * TMath::Pi());
+    else if (mphi > TMath::Pi())
+      mphi -= (2 * TMath::Pi());
+    float dphi = mphi - vphi;
+
+    if (dphi < -1 * TMath::Pi())
+      dphi += (2 * TMath::Pi());
+    else if (dphi > TMath::Pi())
+      dphi -= (2 * TMath::Pi());
+
+    return dphi; // dphi in [-Pi, Pi]
   }
```

Comment:
This code is completely unnecessary. Please consider replacing it with `RecoDecay::constrainAngle(mphi - vphi, -o2::constants::math::PI)`

Reply:
@vkucera thanks for the pointer. I have changed the method here to use the RecoDecay function

Reply:
Thanks